source/common/router/config_impl.h

```diff
@@ -367,8 +367,12 @@ class RouteMatcher {
 
 private:
   const VirtualHostImpl* findVirtualHost(const Http::HeaderMap& headers) const;
+  const VirtualHostSharedPtr findWildcardVirtualHost(const std::string& host) const;
 
   std::unordered_map<std::string, VirtualHostSharedPtr> virtual_hosts_;
+  // std::greater as a minor optimization to iterate from more to less specific
+  std::map<int64_t, std::unordered_map<std::string, VirtualHostSharedPtr>, std::greater>
+      wildcard_virtual_host_suffixes_;
```

Comment:
If this is on the data path and frequently looked up but infrequently changed a sorted array binary searched can have better cache behavior compared to std::map.  This is a potential improvement you could benchmark for v2.